src/scripts/tasks.coffee

```diff
@@ -81,3 +83,37 @@ module.exports = (robot) ->
       msg.send "#{displays[...].join("\n\n")}"
     .catch (error) ->
       msg.reply error
+
+  robot.respond /octo deploy (.+) > ([0-9a-zA-Z.-]+) > (.+)/i, (msg) ->
+    params =
+      project: msg.match[1]
+      version: msg.match[2]
+      env: msg.match[3]
+    if !params.project || !params.version || !params.env
+      msg.reply 'can\'t deploy if you don\'t specify all the params'
+      return
+
+    msg.send "finding project #{params.project}"
+    slugName = OctoHelpers.slugIt params.project
+    client.resources.projects.id(slugName).get()
+    .then(HttpHelpers.validateSuccess)
+    .then(HttpHelpers.getBody)
+    .then(DeployHelper.getProjectId)
+    .then (projectId) ->
+      releaseRequest = client.resources.projects.id(projectId).releases.get()
+      environmentsRequest = client.resources.environments.get()
+      preparedPromises = [releaseRequest, environmentsRequest].map (request) ->
+        request.then(HttpHelpers.validateSuccess).then(HttpHelpers.getBody)
+      Promise.all(preparedPromises)
+    .then (responses) ->
+      deploymentParams =
+        ReleaseId:  DeployHelper.getReleaseId responses[0], params.version
+        EnvironmentId: DeployHelper.getEnvironmentId responses[1], params.env
+        Comments: "Deploy by hubot on behalf of #{msg.message.user.name}"
+    .then(client.resources.deployments.post)
```

Comment:
`.then(client.resources.deployments.post)`

the previous promise just builds the payload.

I'll make that more clear.